File: web/HIGN/HIGN/Personnel/Infrastructure/Persistence/EFC/Repositories/ExaminerRepository.cs
using HIGN.Personnel.Domain.Model.Aggregates;
using HIGN.Personnel.Domain.Repositories;
using HIGN.Shared.Infrastructure.Persistence.EFC.Configuration;
using HIGN.Shared.Infrastructure.Persistence.EFC.Repositories;

namespace HIGN.Personnel.Infrastructure.Persistence.EFC.Repositories;

public class ExaminerRepository(AppDbContext context) : BaseRepository<Examiner>(context), IExaminerRepository
{
    public bool ExistsByNationalProviderIdentifier(string nationalProviderIdentifier)
    {
        return context.Set<Examiner>().AsEnumerable().Any(examiner => examiner.NationalProviderIdentifier == nationalProviderIdentifier);
    }
}

File: openfinalfinal/u202120418/u202120418/src/main/java/org/hign/platform/u202120418/personnel/infrastructure/persistence/jpa/repositories/ExaminerRepository.java
package org.hign.platform.u202120418.personnel.infrastructure.persistence.jpa.repositories;

import org.hign.platform.u202120418.personnel.domain.model.aggregates.Examiner;
import org.hign.platform.u202120418.personnel.domain.model.valueobjects.NationalProviderIdentifier;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

/**
 * ExaminerRepository - Repository for Examiner
 * @author Fiorella Jarama Peñaloza
 * @version 1.0
 */
@Repository
public interface ExaminerRepository extends JpaRepository<Examiner, Long> {
    boolean existsByNationalProviderIdentifier(NationalProviderIdentifier nationalProviderIdentifier);
    Optional<Examiner> findByNationalProviderIdentifier(NationalProviderIdentifier examinerNationalProviderIdentifier);
}
